ui/src/app/utils/timeSpan.ts

```diff
@@ -0,0 +1,42 @@
+import type { Duration } from "date-fns";
+
+import type { Minutes, Seconds, TimeSpan } from "app/base/types";
+
+export const timeSpanToDuration = (timeSpan: TimeSpan | null): Duration => {
+  if (!timeSpan) {
+    return {};
+  }
+  return {
+    hours: Number(timeSpan.match(/([\d]+)\s*(?:hs?|hours?)\s*/)?.[1]),
+    minutes: Number(timeSpan.match(/([\d]+)\s*(?:ms?|mins?|minutes?)\s*/)?.[1]),
+    seconds: Number(timeSpan.match(/([\d]+)\s*(?:s|secs?|seconds?)\s*/)?.[1]),
```

Comment:
This could have been done in other ways, e.g. with regex capture groups, but I decided this is a reasonable compromise for better readability (Regex is confusing enough).

Reply:
There is 1 package[1] that I found that implements parsing time spans, but it seemed unreasonably large for what it does. 

https://github.com/gilly3/timespan-parser

Reply:
Yeah this seems fine to me